bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/Activator.java

```diff
@@ -32,4 +54,33 @@ public void stop(BundleContext context) throws Exception {
     public static BundleContext getContext() {
         return Activator.context;
     }
+
+    /**
+     * Returns the {@link ThingTypeRegistry}
+     *
+     * @return {@link ThingTypeRegistry} or null
+     */
+    public static ThingTypeRegistry getThingTypeRegistry() {
+        return Activator.thingTypeRegistry;
+    }
+
+    private ServiceBinder channelTypeServiceBinder;
+
+    private ServiceBinder thingTypeServiceBinder;
+
+    @Override
+    public void start(BundleContext context) throws Exception {
+        Activator.context = context;
+        thingTypeServiceBinder = new ServiceBinder(context, new ThingTypeRegistryBinder());
+        thingTypeServiceBinder.open();
+        channelTypeServiceBinder = new ServiceBinder(context, new ChannelTypeRegistryBinder());
+        channelTypeServiceBinder.open();
+    }
+
+    @Override
+    public void stop(BundleContext context) throws Exception {
+        channelTypeServiceBinder.close();
```

Comment:
> If I am doing a "activatorMember = new byte[10010241024]" in the start method and does not assign null to activatorMember in the stop method, IMHO the 100MB will not released as long as the activator object is still referenced by the OSGi framework.

Yes, I would assume this is how it works. So if you really claim so much memory, of course it´s better to release it explicitly.  But I think this is not dedicated to OSGi specifically. Also in Servlets or in any other framework you never know when the framework releases the reference to your classes.